protocol/contracts/beanstalk/AppStorage.sol

```diff
@@ -60,20 +69,50 @@ contract Account {
     // The global AppStorage state stores a mapping from account address to Account.State.
     struct State {
         Field field; // A Farmer's Field storage.
-        AssetSilo bean; // A Farmer's Unripe Bean Deposits only as a result of Replant (previously held the V1 Silo Deposits/Withdrawals for Beans).
-        AssetSilo lp;  // A Farmer's Unripe LP Deposits as a result of Replant of BEAN:ETH Uniswap v2 LP Tokens (previously held the V1 Silo Deposits/Withdrawals for BEAN:ETH Uniswap v2 LP Tokens).
-        Silo s; // A Farmer's Silo storage.
+
+        /*
+         * @dev (Silo V1) A Farmer's Unripe Bean Deposits only as a result of Replant
+         *
+         * Previously held the V1 Silo Deposits/Withdrawals for Beans.
+
+         * NOTE: While the Silo V1 format is now deprecated, this storage slot is used for gas
+         * efficiency to store Unripe BEAN deposits. See {FIXME(doc)} for more.
+         */
+        AssetSilo bean; 
+
+        /*
+         * @dev (Silo V1) Unripe LP Deposits as a result of Replant.
+         * 
+         * Previously held the V1 Silo Deposits/Withdrawals for BEAN:ETH Uniswap v2 LP Tokens.
+         * 
+         * BEAN:3CRV and BEAN:LUSD tokens prior to Replant were stored in the Silo V2
+         * format in the next storage slot, `s.a[account].silo`.
+         *
+         * NOTE: While the Silo V1 format is now deprecated, this storage slot is used for gas
+         * efficiency to store Unripe BEAN:3CRV deposits. See {FIXME(doc)} for more.
+         */
+        AssetSilo lp; 
+
+        /*
+         * @dev Silo V2
+         */
+        Silo s;
+        
+        /*
+         * @notice DEPRECATED
+         * 
+         * @dev FIXME(doc)
+         */
         uint32 votedUntil; // DEPRECATED – Replant removed on-chain governance including the ability to vote on BIPs.
         uint32 lastUpdate; // The Season in which the Farmer last updated their Silo.
         uint32 lastSop; // The last Season that a SOP occured at the time the Farmer last updated their Silo.
         uint32 lastRain; // The last Season that it started Raining at the time the Farmer last updated their Silo.
-        uint32 lastSIs; // DEPRECATED – In Silo V1.2, the Silo reward mechanism was updated to no longer need to store the number of the Supply Increases at the time the Farmer last updated their Silo.
-        uint32 proposedUntil; // DEPRECATED – Replant removed on-chain governance including the ability to propose BIPs.
+        uint128 deltaRoots; // the number of roots to add, in the case where a farmer has mowed in the morning 
```

Comment:
in the EVM, data are stored in 32 byte slots. This allows for packing multiple variables into 1 slot, and adding more variables without overwriting the storage slot, as long the new variables do not exceed the 32 byte length. 

In this case, previously 6 uint32s were stored in this slot. This means that 24 bytes were used, and a potential 8 bytes can be added to this slot without affecting the leading slots after. 

by removing the last 2 uint32 variables in the slot, we can now pack a uint128 (16 bytes) into the slot without affecting the leading slots. The slot now looks like: 
uint32  (4 bytes)
uint32  (4 bytes)
uint32  (4 bytes)
uint32  (4 bytes) 
uint128 (16 bytes)
(4+4+4+4+12 = 32) 
additionally, because delta roots is only used in the vesting period, and is always updated before a vesting period plant, the previous values that was contained in the variables will be overwritten and not used.

Reply:
got it. ty for the explanation